doc/https.rst

```diff
@@ -0,0 +1,134 @@
+
+.. code::
+
+    kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.11.0/cert-manager.yaml
+
+For installations of Kubernetes v1.15 or below, you also need to supply
+``--validate=false`` to the above command. For more detail on this, see
+the `Getting Started guide <https://docs.cert-manager.io/en/latest/getting-started/install/kubernetes.html>`_.
+
+We then need to create an issuer that will contact Let's Encrypt for signing
+our certificates. Use the following template to create a new file
+``binderhub-issuer.yaml`` and instantiate it using
+``kubectl apply -f binderhub-issuer.yaml``.
+
+.. code:: yaml
+
+    apiVersion: cert-manager.io/v1alpha2
+    kind: Issuer
+    metadata:
+      name: letsencrypt-production
+      namespace: <same-namespace-as-binderhub>
+    spec:
+      acme:
+        # You must replace this email address with your own.
+        # Let's Encrypt will use this to contact you about expiring
+        # certificates, and issues related to your account.
+        email: <your-contact-mail>
+        server: https://acme-v02.api.letsencrypt.org/directory
+        privateKeySecretRef:
+          # Secret resource used to store the account's private key.
+          name: letsencrypt-production
+        solvers:
+        - http01:
+            ingress:
+              class: nginx
+
+
+Ingress proxy using nginx
+-------------------------
+
+We will use the `nginx ingress controller <https://github.com/kubernetes/ingress-nginx>`_
+to proxy the TLS connection to our BinderHub setup. This will run using
+the static IP we have acquired before. We therefore create a new configuration
+file ``nginx-ingress.yaml``:
+
+.. code:: yaml
+
+    controller:
+      service:
+        loadBalancerIP: <STATIC-IP>
+
+Afterwards we install the ingress proxy using
+``helm install stable/nginx-ingress --name binderhub-proxy --namespace <same-namespace-as-binderhub> -f nginx-ingress.yaml``.
+Then wait until it is ready and showing the correct IP when looking at the output of
+``kubectl --namespace qc-binder-dev get services binderhub-proxy-nginx-ingress-controller``.
+
+Adjust BinderHub config to serve via HTTPS
+------------------------------------------
+
+With the static IP, DNS records and ingress proxy setup, we can now change our
+BinderHub configuration to serve via HTTPS. Therefore adjust your ``config.yaml``
+with the following sections and apply it using ``helm upgrade ...``.
+
+.. code:: yaml
+
+    config:
+      BinderHub:
+        hub_url: https://<jupyterhub-URL>
+
+    jupyterhub:
+      ingress:
+        enabled: true
+        hosts:
+          - <jupyterhub-URL>
+        annotations:
+          kubernetes.io/ingress.class: nginx
+          kubernetes.io/tls-acme: "true"
+          cert-manager.io/issuer: letsencrypt-production
+          https:
+            enabled: true
+            type: nginx
+        tls:
+           - secretName: <jupyterhub-URL>-tls
```

Comment:
Can you use dots `.` in secret names?

Reply:
You can but I think it is not seen as a good practice. Would the following, longer version work better?
```suggestion
           - secretName: <jupyterhub-URL-with-dashes-instead-of-dots>-tls
```

Reply:
Looks good to me.